### PR TITLE
Implement ItemSpawnEvent and ItemDespawnEvent

### PR DESCRIPTION
**Pull request by aramperes**
This PR implements ItemSpawnEvent and ItemDespawnEvent, and replaces GlowItem constructor usage by the `GlowWorld#dropItem` method.

### CODE REVIEW

**Review by kaenganxt — APPROVED** [review]

**Comment by aramperes** [comment]
Do not merge, I found an issue with this but can't fix it for a moment

**Review by mastercoms — APPROVED** [review]

File: src/main/java/net/glowstone/GlowWorld.java [hunk 1]
```diff
@@ -1417,18 +1419,18 @@ public GlowEntity spawn(Location location, Class<? extends GlowEntity> clazz, Sp
...
                     spawnEvent = EventFactory.callEvent(new EntitySpawnEvent(entity));
                 }
-                if (!spawnEvent.isCancelled()) {
-                    List<Message> spawnMessage = entity.createSpawnMessage();
-                    getRawPlayers().stream().filter(player -> player.canSeeEntity(impl)).forEach(player -> player.getSession().sendAll(spawnMessage.toArray(new Message[spawnMessage.size()])));
-                } else {
+                if (spawnEvent != null && spawnEvent.isCancelled()) {
                     // TODO: separate spawning and construction for better event cancellation
                     entity.remove();
+                } else {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/GlowWorld.java:1431]: Shouldn't this branch also require non null? Wrap the whole if else in a null check.

Comment (aramperes) [reply src/main/java/net/glowstone/GlowWorld.java:1431]: No, if the event is null, it means that the event was fired elsewhere. The spawn message is still necessary.

Comment (mastercoms) [reply src/main/java/net/glowstone/GlowWorld.java:1431]: Oh whoops, I don't know how, but I misread that null check as one for `entity`, not `spawnEvent`.

**Review by mastercoms — APPROVED** [review]

**Pull request merged**